chromax/functional.py

```diff
@@ -41,34 +46,44 @@ def cross(
         >>> rec_vec = np.full((n_chr, chr_len), 1.5 / chr_len)
         >>> rec_vec[:, 0] = 0.5  # equal probability on starting haploid
         >>> rec_vec = rec_vec.flatten()
-        >>> random_key = jax.random.PRNGKey(42)
-        >>> f2 = functional.cross(parents, rec_vec, random_key)
+        >>> cross_random_key = jax.random.PRNGKey(42)
+        >>> mutate_random_key = jax.random.PRNGKey(42)
+        >>> mutate_probability = 0.001
+        >>> f2 = functional.cross(parents, rec_vec, cross_random_key, mutate_random_key, mutate_probability)
         >>> f2.shape
         (50, 1000, 2)
     """
     parents = parents.reshape(*parents.shape[:3], -1, 2)
-    random_keys = jax.random.split(random_key, num=len(parents) * 2 * parents.shape[3])
-    random_keys = random_keys.reshape(len(parents), 2, parents.shape[3], 2)
-    offsprings = _cross(parents, recombination_vec, random_keys)
+    cross_random_key = jax.random.split(cross_random_key, num=len(parents) * 2 * parents.shape[3])
+    cross_random_key = cross_random_key.reshape(len(parents), 2, parents.shape[3], 2)
+
+    mutate_split_key = jax.random.split(mutate_split_key, num=len(parents) * 2 * parents.shape[3])
+    mutate_split_key = mutate_split_key.reshape(len(parents), 2, parents.shape[3], 2)
+
+    offsprings = _cross(parents, recombination_vec, cross_random_key, mutate_split_key, mutate_probability)
```

Comment:
I would avoid adding the other keys as argument, but I would generate them internally here.
So you can do:
```python
random_keys = jax.random.split(random_key, num=2 * len(parents) * 2 * parents.shape[3])
random_keys = random_keys.reshape(3, len(parents), 2, parents.shape[3], 2)
cross_random_key, mutate_random_key = random_keys
```

Also, use default value of 0 for mutate_probability, so previous code will continue working

Reply:
random_keys = random_keys.reshape(3, len(parents), 2, parents.shape[3], 2)

you mean 
random_keys = random_keys.reshape(2, len(parents), 2, parents.shape[3], 2)?